### DISCUSSION TITLE
Improve keywordprg for lua filetypes

### DISCUSSION THREAD

**Discussion by craigmac** [neovim/neovim#25781] [Ideas]
Since we've changed the default from Vim to be `:Man`, could we go one step further and do `setlocal keywordprg=:help` on Lua filetypes by default? Using `:Man` for lua doesn't seem useful, and while `K` does have some limitations it will still work for quite a few things in the typical user init file. I ask here rather than upstream Lua ftplugin (I've contributed changes there before) for Vim because it doesn't really apply to Vim. I could pester upstream and make a case for it, since using `man -s` as the default in lua for K isn't very useful anyway.

**Comment by craigmac** [comment]
Also is this the best place to ask things like this or is this a more unofficial channel?

**Reply by dundargoc** [reply]
This is fine.

**Comment by marlun** [comment]
Now `runtime/lua/vim/lsp.lua` checks if `keywordprg` is empty or default before mapping it to LSP hover, as documented in `:h lsp-defaults`. However, if I try to set it to `:help` I'm guessing that is the default value so it's remapped to LSP hover. I would like it to be `:help` in init.lua, any tips on how to do this?

**Reply by marlun** [reply]
I managed to get it working with the following in `after/ftplugin/lua.lua`:
```lua
-- Add mapping to prevent neovim from mapping to LSP hover
vim.keymap.set('n', 'K', 'K', { buffer = true })
-- Set to :help instead of :Man
vim.opt_local.keywordprg = ':help'
```

**Comment by justinmk** [comment]
related:

- https://github.com/neovim/neovim/pull/36205
